kzg/src/eip_7594.rs

```diff
@@ -153,20 +156,45 @@ fn compute_fk20_proofs<
         }
     }
 
-    for i in 0..k2 {
-        h_ext_fft[i] = TG1::g1_lincomb(
-            s.get_x_ext_fft_column(i),
-            &coeffs[i],
-            FIELD_ELEMENTS_PER_CELL,
-            None,
-        );
+    #[cfg(feature = "parallel")]
+    {
+        h_ext_fft.par_iter_mut().enumerate().for_each(|(i, h_ext_fft_elem)| {
+            *h_ext_fft_elem = TG1::g1_lincomb(
+                s.get_x_ext_fft_column(i),
+                &coeffs[i],
+                FIELD_ELEMENTS_PER_CELL,
+                None,
+            );
+        });
+    }
+
+    #[cfg(not(feature = "parallel"))]
+    {
+        for i in 0..k2 {
+            h_ext_fft[i] = TG1::g1_lincomb(
+                s.get_x_ext_fft_column(i),
+                &coeffs[i],
+                FIELD_ELEMENTS_PER_CELL,
+                None,
+            );
+        }
     }
 
     let mut h = vec![TG1::identity(); k2];
     g1_ifft(&mut h, &h_ext_fft, s.get_fft_settings())?;
 
-    for h in h.iter_mut().take(k2).skip(k) {
-        *h = TG1::identity();
+    #[cfg(feature = "parallel")]
+    {
+        h.par_iter_mut().take(k2).skip(k).for_each(|h_elem| {
+            *h_elem = TG1::identity();
+        });
+    }
+
+    #[cfg(not(feature = "parallel"))]
+    {
+        for h_elem in h.iter_mut().take(k2).skip(k) {
+            *h_elem = TG1::identity();
+        }
```

Comment:
This kind of code could be a bit simplified with `kzg::cfg_into_iter` macro (or similar one), so you don't need 2 different implementations, and could just do:
```rs
cfg_iter_mut!(h).take(k2).skip(k).for_each(|h_elem| {
    *h_elem = TG1::identity();
})
```

(`cfg_iter_mut` macro calls `iter_mut`/`par_iter_mut` depending on `parallel` feature)

Reply:
Ok, it is now changed